libflux/src/semantic/parser/mod.rs

```diff
@@ -0,0 +1,1755 @@
+
+        kinds.insert(Kind::Nullable); 
+        bounds.insert(Tvar(0),kinds);
+
+        let pipe_arg = Some(Property{
+            k: "x".to_string(),
+            v: MonoType::Var(Tvar(0)),
+        });
+
+        let output = PolyType {
+            free: vec![Tvar(0)],
+            cons: Some(TvarKinds(bounds)), 
+            expr: MonoType::Fun(Box::new(Function{
+                req: HashMap::new(),
+                opt: HashMap::new(),
+                pipe: pipe_arg,
+                retn: MonoType::Var(Tvar(0))
+            }))
+        };
+
+        assert_eq!(Ok(output), parse_tokens(tokens));
+
+
+        // POLYTYPE: "forall [t0, t1] where t0: Comparable (<-: t0) -> t0"
+        let tokens = vec![
+            Token{token_type: TokenType::FORALL, text: Some("forall".to_string())},
+            Token{token_type: TokenType::LEFTSQUAREBRAC, text: None},
+            Token{token_type: TokenType::IDENTIFIER, text: Some("t0".to_string())},
+            Token{token_type: TokenType::COMMA, text: None},
+            Token{token_type: TokenType::IDENTIFIER, text: Some("t1".to_string())},
+            Token{token_type: TokenType::RIGHTSQUAREBRAC, text: None}, 
+            Token{token_type: TokenType::WHERE, text: Some("where".to_string())},
+            Token{token_type: TokenType::IDENTIFIER, text: Some("t0".to_string())},
+            Token{token_type: TokenType::COLON, text: None},
+            Token{token_type: TokenType::IDENTIFIER, text: Some("Comparable".to_string())},
+            Token{token_type: TokenType::LEFTPAREN, text: None},
+            Token{token_type: TokenType::PIPE, text: None},
+            Token{token_type: TokenType::COLON, text: None},
+            Token{token_type: TokenType::IDENTIFIER, text: Some("t0".to_string())},
+            Token{token_type: TokenType::RIGHTPAREN, text: None},
+            Token{token_type: TokenType::ARROW, text: None},
+            Token{token_type: TokenType::IDENTIFIER, text: Some("t0".to_string())},
+            Token{token_type: TokenType::EOF, text: None}
+        ];
+
+        let mut bounds = HashMap::new();
+        let mut kinds = HashSet::new(); 
+
+        kinds.insert(Kind::Comparable); 
+        bounds.insert(Tvar(0),kinds);
+
+        let pipe_arg = Some(Property{
+            k: String::new(),
+            v: MonoType::Var(Tvar(0)),
+        });
+
+        let output = PolyType {
+            free: vec![Tvar(0), Tvar(1)],
+            cons: Some(TvarKinds(bounds)), 
+            expr: MonoType::Fun(Box::new(Function{
+                req: HashMap::new(),
+                opt: HashMap::new(),
+                pipe: pipe_arg,
+                retn: MonoType::Var(Tvar(0))
+            }))
+        };
+
+        assert_eq!(Ok(output), parse_tokens(tokens));
+    }
+
+    #[test]
+    fn parse_row_polytypes() {
+        let parse_text = "   forall [t1, t2] where t1: Nullable, t2: Comparable {test: t1, testAgain: bool, testLast: [uint]} ";
```

Comment:
Just to reiterate this row type should be specified as:

    {test: t1 | testAgain: bool | testLast: [uint]}